patroni/ha.py

```diff
@@ -1980,8 +2003,9 @@ def get_failover_candidates(self, exclude_failover_candidate: bool) -> List[Memb
         exclude = [self.state_handler.name] + ([failover.candidate] if failover and exclude_failover_candidate else [])
 
         def is_eligible(node: Member) -> bool:
-            # TODO: allow manual failover (=no leader specified) to async node
-            if self.sync_mode_is_active() and not self.cluster.sync.matches(node.name):
+            # in synchronous mode we allow failover (not switchover!) to async node
+            if self.sync_mode_is_active() and not self.cluster.sync.matches(node.name)\
+                    and not (failover and not failover.leader):
```

Comment:
Maybe we could implement a couple helper methods on `Failover` class to help deciding what is going on? I mean something like:

* `def is_switchover...`
* `def is_failover...`

Or some other standardized way that would help whenever we want to check these stuffs?

Reply:
probably, but I am already working on some refactoring of the switchover/failover api in another branch, so maybe we move this suggestion there?

Reply:
Makes sense too, wfm.